### PR TITLE
Adds commentDepth module that lets users set a default and custom depths for comments. fixes #890

### LINKED ISSUE #890: Add depth preference to Hide Child Comments
http://www.reddit.com/r/Enhancement/comments/22acnr/feature_request_ability_to_autohide_comments_that/#

global and, if you're feeling ambitious, per-subreddit overrides

### PR DESCRIPTION
**Pull request by bmalnad**
The module allows users to set a default depth and custom depths by subreddit.

### CODE REVIEW

**Comment by erikdesjardins** [comment]
Looks good, aside from a few nits.

FYI if you add [some variation of](https://github.com/blog/1506-closing-issues-via-pull-requests) "fixes #890" to the description, it'll automatically be closed when this is merged.

File: lib/modules/commentDepth.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			description: 'Enter any subreddit (no "/r/", just the name).  Enter a number for the desired default comment depth. '
+		}
+	};
+
+	// See RESUtils.pageType (utils.js) for other page types
+	module.include = [
+		'all'
+	];
+	module.exclude = [
+	];
```
Comment (erikdesjardins) [inline lib/modules/commentDepth.js]: All pages are included (and none excluded) by default, so these can be omitted.

File: lib/modules/commentDepth.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	module.go = function() {
+		if ((module.isEnabled()) && (module.isMatchURL())) {
+			var commentDepth = parseInt(module.options.defaultCommentDepth.value, 10);
+			if (commentDepth >= 0) {
+				$(document.body).on('click', 'a[href*="/comments"]', function(e) {
+					var href = e.currentTarget.href;
+					if (href.match(RESUtils.regexes.comments)) { 
+						//check for subreddit specific values
+						var subredditCommentDepths = module.options.subredditCommentDepths.value;
+						if(subredditCommentDepths.length > 0){
```
Comment (erikdesjardins) [inline lib/modules/commentDepth.js]: Minor nit: this check is redundant

File: lib/modules/commentDepth.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+								}
+							}				    		
+						}
+						//check for ? in url
+						if (href.match(/\?/)){
+							//check for 'depth='
+							href = (href.match(/depth=/)) ? href : href + '&depth=' + commentDepth;
+						}else{
+							href += '?depth=' + commentDepth;
+						}
```
Comment (erikdesjardins) [inline lib/modules/commentDepth.js]: Might as well bail-out earlier (i.e. on line 40) if `href.match(/[?&]depth=/)`, then `href = RESUtils.insertParam(href, 'depth', commentDepth)` can be used here.

**Comment by bmalnad** [comment]
I cleaned up the issues you mentioned.  Let me know if you notice anything else.  Thanks!

**Comment by andytuba** [comment]
This looks pretty awesome, thanks for contributing!

My one comment is: how do we feel about this being enabled by default? This seems like a nice power-user tweak but I don't feel the need to override reddit's default depth algorithm.

File: lib/modules/commentDepth.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+							return;
+						}
+						//check for subreddit specific values
+						var subredditCommentDepths = module.options.subredditCommentDepths.value;
+						for (var i = 0; i < subredditCommentDepths.length; i++) {
+							if(href.indexOf('/r/'+subredditCommentDepths[i][0]+'/') >= 0){
+								commentDepth = subredditCommentDepths[i][1];
+								break;
+							}
+						};
```
Comment (andytuba) [inline lib/modules/commentDepth.js]: lol linter..

**Comment by andytuba** [comment]
:warning: TODO: add module to other manifests.  (I could take care of that while merging.)

**Comment by bmalnad** [comment]
Disabling it by default is probably a good idea.

**Pull request closed without merging**